src/android/TTS.java

```diff
@@ -51,51 +53,101 @@ public class TTS extends CordovaPlugin implements OnInitListener {
     public static final String ERR_ERROR_INITIALIZING = "ERR_ERROR_INITIALIZING";
     public static final String ERR_UNKNOWN = "ERR_UNKNOWN";
 
-    boolean ttsInitialized = false;
     TextToSpeech tts = null;
     Context context = null;
+    CordovaWebView webViewContext = null;
+    CallbackContext execCallbackContext = null;
+
+    String textToRead;
+    String locale;
+    double speechRate = 1.0;
+    double pitch = 1.0;
+    boolean cancel = false;
+    String identifier;
+
+    private UtteranceProgressListener TTSListener = new UtteranceProgressListener() {
+        @Override
+        public void onStart(String s) {
+            // do nothing
+        }
 
-    @Override
-    public void initialize(CordovaInterface cordova, final CordovaWebView webView) {
-        context = cordova.getActivity().getApplicationContext();
-        tts = new TextToSpeech(cordova.getActivity().getApplicationContext(), this);
-        tts.setOnUtteranceProgressListener(new UtteranceProgressListener() {
-            @Override
-            public void onStart(String s) {
-                // do nothing
+        @Override
+        public void onDone(String callbackId) {
+            resetSpeechParams();
+            if (!callbackId.equals("")) {
+                CallbackContext context = new CallbackContext(callbackId, webViewContext);
+                context.success();
             }
+        }
 
-            @Override
-            public void onDone(String callbackId) {
-                if (!callbackId.equals("")) {
-                    CallbackContext context = new CallbackContext(callbackId, webView);
-                    context.success();
-                }
+        @Override
+        public void onError(String callbackId) {
+            resetSpeechParams();
+            if (!callbackId.equals("")) {
+                CallbackContext context = new CallbackContext(callbackId, webViewContext);
+                context.error(ERR_UNKNOWN);
             }
+        }
+    };
 
-            @Override
-            public void onError(String callbackId) {
-                if (!callbackId.equals("")) {
-                    CallbackContext context = new CallbackContext(callbackId, webView);
-                    context.error(ERR_UNKNOWN);
-                }
-            }
-        });
+
+    @Override
+    public void initialize(CordovaInterface cordova, final CordovaWebView webView) {
+        context = cordova.getActivity().getApplicationContext();
+        webViewContext = webView;
+        try {
+            createTTSInstance(execCallbackContext);
```

Comment:
`execCallbackContext` will always be null on initialize, so should probably replace it with `null` for this specific call just to make the intention more clear